src/MetadataManager/pathToHash.ts

```diff
@@ -0,0 +1,280 @@
+            const obj = queue.pop();
+            if(obj===undefined) {
+                continue;
+            }
+            if(vscode.workspace.workspaceFolders === undefined) {
+                break;
+            }
+            let path = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri,obj[2]).path;
+            if(fs.lstatSync(path).isDirectory()){
+                for(let key in obj[1]){
+                    queue.push([key, obj[1][key], obj[2]+"/"+key]);
+                }
+            }else{
+                flatPathTohash[obj[2]] = obj[1];
+            }
+        }
+        return flatPathTohash;
+    }
+
+    public async createMetadata (flattenPathToHash: any){
+        
+        for(let path in flattenPathToHash){
+            const hash = flattenPathToHash[path];
+
+            let metadata: any = await Metadata.getMetadata(hash);
+    
+            if(Object.keys(metadata).length === 0){
+                await Metadata.setMetadata(hash, {});
+
+            }else if(metadata[path]){
+                return;
+            }
+            
+            const filename: any = path.split('/').pop();
+            if(vscode.workspace.workspaceFolders === undefined) {
+                return;
+            }
+            const stats: any = await MetadataEventManager.getStats(vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri, path));
+            metadata[path] = {};
+            metadata[path]["name"] = filename;
+            metadata[path]["fileExtension"] = filename.split(".")[1];
+            metadata[path]["createTime"] = stats.birthtime;
+            metadata[path]["modifiedTime"] = stats.mtime;
+            metadata[path]["isDeleted"] = false;
+            await Metadata.setMetadata(hash, metadata);  
+        }
+    }
+
+    public async deleteMetadata(flattenpathToHash : any){
+        if(vscode.workspace.workspaceFolders===undefined) {
+            return;
+        }
+        const baseUri = vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0].uri,".meta/hash_objects");
+        const arrayList = await vscode.workspace.fs.readDirectory(baseUri);
+        for(const array of arrayList){
+            const hashFolderUri = vscode.Uri.joinPath(baseUri, array[0]);
+            const hashList = await vscode.workspace.fs.readDirectory(hashFolderUri);
+            for(const hashFile of hashList){
+                const hashUri = vscode.Uri.joinPath(hashFolderUri, hashFile[0]);
+                let metadata = JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(hashUri)).toString());
+                const hash = array[0] + hashFile[0].split('.')[0];
+                for(const key in metadata){
+                    if(!metadata[key].isDeleted && flattenpathToHash[key] !== hash){
+                        metadata[key].isDeleted = true;
+                    }
+                }
+                Metadata.setMetadata(hash, metadata);
+            }
+        }
+    }
+
+
+    public getPathToHash(uri: vscode.Uri ) {
```

Comment:
```suggestion
    public get(uri: vscode.Uri ) {
```


There is no need to repeat the classname 'PathToHash' (ditto)